### ISSUE TITLE
Test rustup on cross-compile targets

### ISSUE THREAD

**Issue by brson** [rust-lang-nursery/rustup.rs#690]
People depend on rustup for all sorts of hardware that I don't have, and that isn't tested. It would be nice to verify that it actually works. Best to come up with some solution that works in a Docker container and can run `cargo test` against an emulator, probably qemu.

cc https://github.com/rust-lang-nursery/rustup.rs/issues/687

@japaric Do you have a solution for running tests against emulators for common targets?

**Comment by japaric** [comment]
Yeah, I've been using [this docker image](https://hub.docker.com/r/japaric/rustc-builtins/) ([Dockerfile](https://github.com/japaric/docker/tree/master/rustc-builtins)) to test [rustc-builtins](https://github.com/japaric/rustc-builtins), a Rust port of compiler-rt builtins, on almost (\*) [every single arch](https://travis-ci.org/japaric/rustc-builtins) that Rust supports via QEMU emulation.

I've been meaning to test integrating the approach used in that repo into the rust-lang/libc repo but I've been busy lately. I should have time to look into it soon though. If I have extra time, I can look into integrating that into this repository as well.

(\*) Can't test on ppc64le because QEMU for that arch appears to be broken; it segfaults even running this simple C program: `int main() { return 0; }`.

**Comment by japaric** [comment]
Oh, and I don't recommend using that docker image to build binary releases. The image uses Ubuntu 16.04 and its cross toolchains come with glibc-2.2x. Using those toolchains would generate a binary that depends on a too new glibc version, which would make the binaries less portable.
